src/main/java/org/stellar/base/AssetTypeNative.java

```diff
@@ -7,6 +7,16 @@
   public AssetTypeNative() {}
 
   @Override
+  public String getCode() {
```

Comment:
I'm concerned that we're painting ourselves into a corner here.  We cannot assume that all currencies going forward will have the concept of a Code and an Issuer, so IMO baking it into the core `Asset` contract is going to bite us in the ass.